### PR TITLE
SRS OC BODBAW - improvements for absorption

### PR DESCRIPTION
**Pull request by lbesnard**
* xls sheet ap were not used
 * var position for absorption are found in a smarter way

### CODE REVIEW

File: SRS/SRS_OC_BODBAW/srs_oc_bodbaw_netcdf_creation.py [hunk 1]
```diff
@@ -219,19 +219,22 @@ def data_frame_ac9_hs6(self):
...
-        data_frame.Longitude    = self.sheet.row_at(self.idx_start_data + 4)[2:self.max_data_col + 2]
-        data_frame.Depth        = self.sheet.row_at(self.idx_start_data + 5)[2:self.max_data_col + 2]
+        data_frame.Station_Code = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Station_Code'))[2:self.max_data_col + 2]
+        data_frame.Latitude     = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Latitude'))[2:self.max_data_col + 2]
+        data_frame.Longitude    = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Longitude'))[2:self.max_data_col + 2]
+        data_frame.Depth        = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Depth'))[2:self.max_data_col + 2]
+        if 'Sample_Number' in self.sheet.column_at(0)[self.idx_start_data:]:
+            data_frame.Sample_Number = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Sample_Number'))[2:self.max_data_col + 2]
```
Comment (mhidas) [inline SRS/SRS_OC_BODBAW/srs_oc_bodbaw_netcdf_creation.py]: You could make these lines slightly shorter and easier to read by assigning 
```python
self.sheet.column_at(0)[self.idx_start_data:]
```
to a local variable with a shorter name?

**Comment by lbesnard** [comment]
thanks @mhidas . Just done it. I'm happy to rebase afterwards

File: SRS/SRS_OC_BODBAW/srs_oc_bodbaw_netcdf_creation.py [hunk 2]
```diff
@@ -219,24 +219,27 @@ def data_frame_ac9_hs6(self):
...
 
-        data_frame.Station_Code = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Station_Code'))[2:self.max_data_col + 2]
-        data_frame.Latitude     = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Latitude'))[2:self.max_data_col + 2]
-        data_frame.Longitude    = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Longitude'))[2:self.max_data_col + 2]
-        data_frame.Depth        = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Depth'))[2:self.max_data_col + 2]
+        data_frame.Station_Code = [self.sheet.row_at(self.idx_start_data + col0_data.index('Station_Code'))[i] for i in idx_col_val]
+        data_frame.Latitude     = [self.sheet.row_at(self.idx_start_data + col0_data.index('Latitude'))[i] for i in idx_col_val]
+        data_frame.Longitude    = [self.sheet.row_at(self.idx_start_data + col0_data.index('Longitude'))[i] for i in idx_col_val]
+        data_frame.Depth        = [self.sheet.row_at(self.idx_start_data + col0_data.index('Depth'))[i] for i in idx_col_val]
```
Comment (mhidas) [inline SRS/SRS_OC_BODBAW/srs_oc_bodbaw_netcdf_creation.py]: `col0_data` definitely helps, but for me this list comprehension with `idx_col_val` is neither shorter nor clearer, so I'm not sure why you would do it.

**Comment by mhidas** [comment]
Go ahead and rebase. I'll merge it.

**Comment by lbesnard** [comment]
@mhidas done !

**Pull request merged**